samples/03.live-inking/src/main-view.ts

```diff
@@ -0,0 +1,334 @@
+            <div id="inkingHost"></div>
+        </div>
+        <div id="buttonStrip">
+            <div class="toolbar">
+                <button id="btnStroke">Stroke</button>
+                <button id="btnLaserPointer">Laser pointer</button>
+                <button id="btnHighlighter">Highlighter</button>
+                <button id="btnEraser">Eraser</button>
+                <button id="btnPointEraser">Point eraser</button>
+                <button id="btnClear" style="margin-left: 20px;">Clear</button>
+                <button id="btnToggleCursorShare">Share cursor</button>
+            </div>
+            <div class="toolbar">
+                <button id="btnBlack">Black</button>
+                <button id="btnRed">Red</button>
+                <button id="btnGreen">Green</button>
+                <button id="btnBlue">Blue</button>
+                <button id="btnYellow">Yellow</button>
+            </div>
+            <div class="toolbar">
+                <button id="btnZoomOut">Zoom out</button>
+                <button id="btnZoomIn">Zoom in</button>
+                <button id="btnOffsetLeft" style="margin-left: 20px;">Offset left</button>
+                <button id="btnOffsetUp">Offset up</button>
+                <button id="btnOffsetRight">Offset right</button>
+                <button id="btnOffsetDown">Offset down</button>
+                <button id="btnResetView" style="margin-left: 20px;">Reset view</button>
+            </div>
+            <div class="toolbar">
+                <button id="btnSimulation">Start simulation</button>
+                <button id="btnOpenNewWindow">Open new window</button>
+            </div>
+        </div>
+    </div>`;
+
+const containerSchema = {
+    initialObjects: {
+        liveCanvas: LiveCanvas,
+        startStopDrawingSimulation: EphemeralEvent
+    }
+};
+
+export class MainView extends View {
+    private _inkingManager!: InkingManager;
+    private _container!: IFluidContainer;
+    private _drawingSimulation!: DrawingSimulation;
+    private _simulationStarted = false;
+
+    private offsetBy(x: number, y: number) {
+        this._inkingManager.offset = {
+            x: this._inkingManager.offset.x + x,
+            y: this._inkingManager.offset.y + y
+        }
+
+        this.updateBackgroundImagePosition();
+    }
+
+    private getLiveCanvas(): LiveCanvas {
+        return this._container.initialObjects.liveCanvas as LiveCanvas;
+    }
+
+    private startOrStopDrawingSimulation(start: boolean) {
+        (this._container.initialObjects.startStopDrawingSimulation as EphemeralEvent).sendEvent({ isStarted: start });
+    }
+
+    private runningInTeams(): boolean {
+        const params = new URLSearchParams(window.location.search);
+        const config = params.get("inTeams");
+
+        return config !== null && config.toLowerCase() === "1";
+    }
+
+    private _hostResizeObserver!: ResizeObserver;
```

Comment:
is `!` needed?

Reply:
It is because I'm not initializing it in the constructor.